### DISCUSSION TITLE
year_vtg and year_act in ACT | Interpretation of ACT

### DISCUSSION THREAD

**Discussion by shreeyashn20** [iiasa/message_ix#679] [General]
Dear Fellow Researcher,

After running the model, I have extracted ACT as follows:
t | n | yv | ya | lvl
elec_pump | India | 2020 | 2020 | 10.44
elec_pump | India | 2020 | 2025 |   1.24
elec_pump | India | 2025 | 2025 | 14.56
elec_pump | India | 2025 | 2030 |   6.88
elec_pump | India | 2030 | 2030 | 18.23

Now, first row having same year_vtg and year_act i.e., 2020 and 2020 respectively. I assume this technology meets the demand starting its operation from 01-01-2020 to 31-12-2025. As per my interpretation, second row although has vintage year of 2020 but the operation of technology starts in 2025 say 01-01-2025. However in the third row, same technology is also starting its operation from 01-01-2025. Third row is exactly equal to demand for 2025 i.e., 14.56. Then my question is - What does second row indicates? What is the use of additional activity in second row? For final activity should I do cumulative activity as per vintage year basis or activity year basis. 

Please help me in interpretation of ACT and other outputs like PRICE_COMMODITY, EXTR, etc. 

Thank you!

**Comment by Jihoon** [comment]
Dear @shreeyashn20, you are right in interpreting that `year_vtg` is when the capacity of the technology is installed, and that `year_act` is when the capacity is operated and generates output. But I see you say "has the vintage year of 2020 but the operation of technology **starts** in 2025". To be precise, it is not really '**starting**' the activity from 01-01-2025, but actually '**reducing**' the activity from 10.44 (2020) to 1.24 (2025) for some reason. (One reason can be that your technical_lifetme is shorter than 5x2=10 years?)

Anyway for `year_act=2025` in your setting, total output is generated both by the capacity built in 2020 and 2025 periods. And if I can assume the `output` parameter is 1, the period 2025 will see the output of `14.56+1.24=15.80`. 

But I guess you meant that the external demand for the output of this technology for 2025 is only 14.56. In that case, the extra 1.24 can be due to other factors, e.g. unintended constraints or connections to other technologies etc. For the accounting of the activity, it really depends on what you want to see in the end. Total output from the technology in each period should be based on `year_act`.
